app/mutations/addresses/request.rb

```diff
@@ -63,10 +63,10 @@ def request_static
     def request_dynamic
       available = @pool.available_addresses
 
-      info "pool #{@pool} allocates from #{@pool.allocation_range} and has #{available.length} available addresses"
+      info "pool #{@pool} allocates from #{@pool.allocation_range} and has #{available.size} available addresses"
 
       # allocate
-      unless allocate_address = policy.allocate_address(available)
+      unless allocate_address = policy.allocate_address(available.first(100).to_a)
```

Comment:
Not really sure but some part of my brain would like to see this `available.first(100).to_a` part handled within the `policy.allocate_address`. The other part is fine with this. :)